### ISSUE TITLE
azuread_application data source 'oauth2Permissions'

### ISSUE THREAD

**Issue by mion00** [terraform-providers/terraform-provider-azuread#67]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I would like to have some additional details from the application manifest, in particular the oauth2Permissions ID. I need it to delegate access to another application I am creating as part of Terraform, as described in the official guide for AKS integration with AD, in the [section](https://docs.microsoft.com/en-us/azure/aks/aad-integration#create-client-application) related to the client application, because it needs to delegate access to the server application created before.

The workaround I found is to query with az cli the necessary field, save it as an external data source. I think a more elegant solution would be to get the field directly in terraform.

### New or Affected Resource(s)

* azuread_application

### Potential Terraform Configuration

```hcl
# AD server application
resource "azuread_application" "kubernetes-production-cluster-AADserver" {
  name                       = "kubernetes-production-cluster-AADserver"
  available_to_other_tenants = false

  # oauth2_allow_implicit_flow = true
  required_resource_access {
    resource_app_id = "00000003-0000-0000-c000-000000000000" # Microsoft Graph API

    # Necessary permissions
    resource_access {
      id   = "7ab1d382-f21e-4acd-a863-ba3e13f7da61" # Read directory data
      type = "Role"
    }

    resource_access {
      id   = "e1fe6dd8-ba31-4d61-89e7-88639da4683d" # Sign in and read user profile
      type = "Scope"
    }

    resource_access {
      id   = "06da0dbc-49e2-44d2-8312-53f166ab848a" # Read directory data permission
      type = "Scope"
    }
  }
}

# External data source to get information about the AADserver AD application
data "external" "AADserver_application" {
  program = ["bash", "azure_ad.sh"]

  query = {
    # arbitrary map from strings to strings, passed to the external program as the data query
    application_id = "${azuread_application.kubernetes-production-cluster-AADserver.application_id}"
  }

}

# AD client application
resource "azuread_application" "kubernetes-production-cluster-AADclient" {
  name                       = "kubernetes-production-cluster-AADclient"
  available_to_other_tenants = false

  oauth2_allow_implicit_flow = true

  # Set reply urls, or the login will fail
  reply_urls = ["https://kubernetes-production-cluster-AADclient"]

  # Setup access to the AAD server application
  required_resource_access {
    resource_app_id = "${azuread_application.kubernetes-production-cluster-AADserver.application_id}" # Server AAD application id

    # Necessary permissions
    resource_access {
      id   = "${data.external.AADserver_application.result["oauth2PermissionsID"]}" # OAUTH2 application ID
      type = "Scope"
    }
  }

}
```

**Comment by lzadjsf** [comment]
Any news? It would be very usefull for managing application for AKS cluster deployment.

**Comment by lzadjsf** [comment]
Any news? It would be very usefull for managing application for AKS cluster deployment.

**Comment by jfcoz** [comment]
Seems to be here : https://github.com/terraform-providers/terraform-provider-azuread/pull/79

**Comment by katbyte** [comment]
fixed by #79
